### PR TITLE
Removed parsing pod

### PR DESCRIPTION
**Pull request by saurabh-prakash**
# Description

Removed the parsing mode altogether as it is adding up latency in overall build

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

This has been tested locally on synapse by running a webhook for `express` repo having 2 parallelism

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: cmd/nucleus/bin.go [hunk 1]
```diff
@@ -139,8 +138,6 @@ func run(cmd *cobra.Command, args []string) {
 		logger.Fatalf("failed to initialize cache manager: %v", err)
 	}
 
-	parserService := parser.New(tcm, logger)
```
Comment (palashmarantas) [inline cmd/nucleus/bin.go:142]: why we removing parsing pod ? Please add jira ticket and fill necessary details.

Comment (saurabh-prakash) [reply cmd/nucleus/bin.go:142]: this is still a WIP. Therefore, had marked it as Draft

File: pkg/core/interfaces.go [hunk 2]
```diff
@@ -128,3 +122,7 @@ type ExecutionManager interface {
 	// StoreCommandLogs stores the command logs in the azure.
 	StoreCommandLogs(ctx context.Context, blobPath string, reader io.Reader) <-chan error
 }
+
+type Requests interface {
```
Comment (rishabh-arya95) [inline pkg/core/interfaces.go:126]: nit: add comment

Comment (saurabh-prakash) [reply pkg/core/interfaces.go:127]: Addressed. However, this was supposed to be caught by automated `golangci-lint` checks. Its missing a configuration file. I have added that in this diff but eventually commented it out as all/any existing linting issues need to be fixed as a separate PR.

**Review by rishabh-arya95 — APPROVED** [review]

**Review by utkarsh-lambdatest — APPROVED** [review]
LGTM

**Pull request merged**